inference_perf/datagen/hf_billsum_datagen.py

```diff
@@ -0,0 +1,107 @@
+        else:
+            raise ValueError(f"Invalid dataset path: No dataset path provided")
+
+        self.min_num_turns = 2
+        self.data_key = "conversations"
+        self.role_key = "from"
+        self.content_key = "value"
+        
+        # Advance the iterator to the first data point
+        next(self.billsum_dataset)
+
+    def get_supported_apis(self) -> List[APIType]:
+        return [APIType.Chat, APIType.Completion]
+
+    def get_data(self) -> Generator[InferenceAPIData, None, None]:
+        if self.billsum_dataset is not None:
+            while True:
+                data = next(self.billsum_dataset)
+                if (
+                    data is None
+                    or data[self.data_key] is None
+                    or len(data[self.data_key]) < self.min_num_turns
+                    or len(data[self.data_key]) == 0
+                ):
+                    continue
+
+                if self.api_config.type == APIType.Completion:
+                    try:
+                        prompt = data[self.data_key][0].get(self.content_key)
+                        completion = data[self.data_key][1].get(self.content_key)
+                        if not prompt:
+                            continue
+                        # Ensured by main.py logic and __init__ type hint for this class
+                        assert self.tokenizer is not None
+                        completion_tokens = self.tokenizer.count_tokens(completion)
+                        prompt_tokens = self.tokenizer.count_tokens(prompt)
+
+                        if self.input_distribution:
+                            if prompt_tokens < self.input_distribution.min or prompt_tokens > self.input_distribution.max:
+                                continue
+                        if self.output_distribution:
+                            if (
+                                completion_tokens < self.output_distribution.min
+                                or completion_tokens > self.output_distribution.max
+                            ):
+                                continue
+
+                        yield CompletionAPIData(prompt=prompt, max_tokens=completion_tokens)
+                    except (KeyError, TypeError) as e:
+                        logger.warning(f"Skipping invalid completion data: {e}")
+                        continue
+                elif self.api_config.type == APIType.Chat:
+                    yield ChatCompletionAPIData(
+                        messages=[
+                            ChatMessage(role=conversation[self.role_key], content=conversation[self.content_key])
+                            for conversation in data[self.data_key]
+                        ]
+                    )
+                else:
+                    raise Exception("Unsupported API type")
+
+    def is_io_distribution_supported(self) -> bool:
+        return True
+
+    def is_shared_prefix_supported(self) -> bool:
+        return True
```

Comment:
Should this be false? I don't see any prefix specific parameters being applied to this dataset.